quadratic/quadratic.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+func main() {
+	fmt.Println("The program finds solutions of quadratic equation.")
+	fmt.Print("Enter the coefficients A, B and C: ")
+	var a, b, c float64
+	fmt.Scan(&a, &b, &c)
+	if a == 0 {
+		if b == 0 {
+			if c == 0 {
+				fmt.Println("Infinite roots")
+			} else {
+				fmt.Println("No roots")
+			}
+		} else {
+			x := -c / b
+			fmt.Println("x =", x)
+		}
+		return
+	}
+	d := b*b - 4*a*c
+	if d < 0 {
+		fmt.Println("No roots")
```

Comment:
optional: Technically, this is "no REAL roots"
Square equation ALWAYS has two roots (in some cases first root is exactly equal to the second, and in some cases those are imaginary
I'll give extra points for findings complex roots as well